newdle/api.py

```diff
@@ -307,52 +307,55 @@ def create_newdle(title, duration, timezone, timeslots, participants, private, n
         notify=notify,
     )
     db.session.add(newdle)
+    db.session.flush()
+    send_invitation_emails(newdle)
     db.session.commit()
-    notify_newdle_participants(
-        newdle,
-        f'Invitation: {newdle.title}',
-        'invitation_email.txt',
-        'invitation_email.html',
-        lambda p: {
-            'creator': newdle.creator_name,
-            'title': newdle.title,
-            'participant': p.name,
-            'summary_link': url_for('newdle_summary', code=newdle.code, _external=True),
-            'answer_link': url_for(
-                'newdle', code=newdle.code, participant_code=p.code, _external=True
-            ),
-        },
-    )
     return NewdleSchema().jsonify(newdle)
 
 
-@api.route('/newdle/<code>')
-@allow_anonymous
-def get_newdle(code):
-    newdle = Newdle.query.filter_by(code=code).first_or_404(
-        'Specified newdle does not exist'
-    )
-    if newdle.deleted:
-        return DeletedNewdleSchema().jsonify(newdle)
-    return RestrictedNewdleSchema().jsonify(newdle)
-
-
 @api.route('/newdle/<code>', methods=('PATCH',))
-@use_args(UpdateNewdleSchema(), locations=('json',))
+@use_args(UpdateNewdleSchema(partial=True), locations=('json',))
 def update_newdle(args, code):
     newdle = Newdle.query.filter_by(code=code).first_or_404(
         'Specified newdle does not exist'
     )
-    if newdle.creator_uid != g.user['uid']:
+    if g.user is None or newdle.creator_uid != g.user['uid']:
         raise Forbidden
+    new_participants = []
+    if 'participants' in args:
+        participants = args.pop('participants')
+        # Filter the new participants to be created (excluding anonymous)
+        new_participants = {
+            Participant(**p)
+            for p in participants
+            if 'id' not in p and 'auth_uid' in p and p['auth_uid'] is not None
```

Comment:
```suggestion
            if 'id' not in p and p.get('auth_uid') is not None
```

Reply:
I guess I was focused on the logic too much 🤦‍♂️